spec/checkMergeConflictSpec.js

```diff
@@ -0,0 +1,216 @@
+      expect(
+        checkMergeConflictModule.checkMergeConflictsInAllPullRequests
+      ).toHaveBeenCalled();
+      expect(
+        checkMergeConflictModule.checkMergeConflictsInPullRequest
+      ).toHaveBeenCalled();
+    });
+
+    it('ping pr author regarding merge conflict', () => {
+      expect(github.issues.createComment).toHaveBeenCalled();
+
+      const link = 'link'.link(
+        'https://help.github.com/articles/resolving-a-merge-conflict-using-the-command-line/'
+      );
+      const params = {
+        repo: payload.payload.repository.name,
+        owner: payload.payload.repository.owner.login,
+        number: payload.payload.pull_request.number,
+        body:
+          'Hi @' +
+          payload.payload.pull_request.user.login +
+          '. Due to recent changes in the "develop" branch, ' +
+          'this PR now has a merge conflict. ' +
+          'Please follow this ' +
+          link +
+          ' if you need help resolving the conflict, ' +
+          'so that the PR can be merged. Thanks!',
+      };
+      expect(github.issues.createComment).toHaveBeenCalledWith(params);
+    });
+
+    it('adds merge conflict label', () => {
+      expect(github.issues.addLabels).toHaveBeenCalled();
+      const params = {
+        repo: payload.payload.repository.name,
+        owner: payload.payload.repository.owner.login,
+        number: payload.payload.pull_request.number,
+        labels: [mergeConflictLabel.name],
+      };
+      expect(github.issues.addLabels).toHaveBeenCalledWith(params);
+    });
+  });
+
+  describe("when pull request that doesn't cause merge conflict gets merged", () => {
+    beforeEach(async () => {
+      // Simulate a merged conflict in pull request.
+      payload.payload.pull_request.merged = true;
+      payload.payload.action = 'closed';
+
+      const pullRequest = { ...payload.payload.pull_request };
+      pullRequest.merged = false;
+      pullRequest.mergeable = true;
+
+      github.pulls = {
+        get: jasmine.createSpy('get').and.resolveTo({
+          data: pullRequest,
+        }),
+        list: jasmine.createSpy('list').and.resolveTo({
+          data: [pullRequest],
+        }),
+      };
+
+      await robot.receive(payload);
+    });
+
+    it('checks for merge conflict', () => {
+      expect(
+        checkMergeConflictModule.checkMergeConflictsInAllPullRequests
+      ).toHaveBeenCalled();
+      expect(
+        checkMergeConflictModule.checkMergeConflictsInPullRequest
+      ).toHaveBeenCalled();
+    });
+
+    it('does not ping pr author', () => {
+      expect(github.issues.createComment).not.toHaveBeenCalled();
+    });
+    it('does not add merge conflict label', () => {
+      expect(github.issues.addLabels).not.toHaveBeenCalled();
+    });
+  });
+
+  describe('when pull request already has merge conflict label', () => {
+    beforeEach(async () => {
+      // Simulate a merged conflict in pull request.
+      payload.payload.action = 'synchronize';
```

Comment:
Explain the purpose.

Reply:
Other tests were handling the pull_request.closed event, so I decided to switch up the event here since the synchronize event performs the same action (check for merge conflict).